addons/stock/models/stock_move.py

```diff
@@ -977,15 +977,18 @@ def _assign_picking(self):
             # for each move that why moves[0] is acceptable
             picking = moves[0]._search_picking_for_assignation()
             if picking:
-                if any(picking.partner_id.id != m.partner_id.id or
-                        picking.origin != m.origin for m in moves):
-                    # If a picking is found, we'll append `move` to its move list and thus its
+                # If a picking is found, we'll append `move` to its move list and thus its
+                    # `partner_id` and `ref` field will refer to multiple records. In this
+                    # case, we chose to  wipe them.# If a picking is found, we'll append `move` to its move list and thus its
                     # `partner_id` and `ref` field will refer to multiple records. In this
                     # case, we chose to  wipe them.
-                    picking.write({
-                        'partner_id': False,
-                        'origin': False,
-                    })
+                vals = {}
+                if any(picking.partner_id.id != m.partner_id.id for m in moves):
```

Comment:
I would also add and m.partner_id so we won't remove the partner if the move has no partner.

Reply:
```suggestion
                if any(m.partner_id and picking.partner_id.id != m.partner_id.id for m in moves):
```